File: projects/m2/007-only-the-words/solutions/lisaHu/solution-007-only-the-words.py
def only_the_words(s):
    #create a list of words in string using blank spaces to determine start and end of element
    word_list = s.split(' ')
    new_word_list = []
    punctuation_marks = ['!','?','.',',',':',';','-',"'"]

    #for each word in list, check if first and last character are in the list of puntaction marks list
    for word in word_list:

        #if they are, replace them with blank (nothing) and append the new word in new list
        #if they aren't append the word in new list as there is nothing to chnge
        if word[-1] in punctuation_marks or word[0] in punctuation_marks:
            if word[-1] in punctuation_marks:
                word = word.replace(word[-1], '')
            if word[0] in punctuation_marks:
                word = word.replace(word[0], '')
            new_word_list.append(word)
        else:
            new_word_list.append(word)
    return new_word_list

#get string from user and print return of function
string = input('Enter a string: ')
print(*only_the_words(string), sep = ' ')

File: projects/m2/007-only-the-words/solutions/EvaLateltin/solution-007-only-the-words.js
/*
LOGIC
Create an array of all punctuation marks
-> commas, periods, question marks, hyphens, apostrophes, exclamation points, colons, and semicolons.
Create a prompt function that gets an input from the user. 
Split the string into an array - the words will be divided by the blank space. 
if a word starts or ends with a punctuation mark, remove it.
Check if the first or last digit of a String is a punctuation mark and remove it.
Return the cleaned array
*/

const punctuationMarks = [",", ".", "?", "-", "'", "!", ":", ";"];

const userInput = prompt("Enter your sentence.");

const userInputArray = userInput.split(" ");

const userInputCleanedArray = [];

userInputArray.forEach((word) => {
  for (punctuationMark of punctuationMarks) {
    if (word.startsWith(punctuationMark)) {
      word = word.substring(1, word.length);
    }
    if (word.endsWith(punctuationMark)) {
      word = word.substring(0, word.length - 1);
    }
  }
  userInputCleanedArray.push(word);
});

alert(userInputCleanedArray);
